int8 obstacle_distance
int8 turn_direction